kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -55,34 +54,35 @@ private val ABSENT_VALUE = Any()
  * constructor, and then by setting any additional properties that exist, if any.
  */
 internal class KotlinJsonAdapter<T>(
-    val constructor: KFunction<T>,
-    val bindings: List<Binding<T, Any?>?>,
-    val options: JsonReader.Options) : JsonAdapter<T>() {
+  val constructor: KFunction<T>,
+  val allBindings: List<Binding<T, Any?>?>,
+  val nonTransientBindings: List<Binding<T, Any?>>,
```

Comment:
I find this more difficult to follow now. Can we accomplish the same work with a single list still?

Reply:
I've been fiddling with this locally, and I don't think there's a way to get a single list here without making the JsonAdapterFactory logic more convoluted than just having this list is. I'd like to suggest we rebase this PR and merge with the current logic to get the issue resolved, and can revisit the readability separately. WDYT?